src/ast/nodes.hpp

```diff
@@ -329,43 +330,20 @@ struct TranslationUnit : public AstNode<TranslationUnit> {
   inline bool isDecorated() const { return true; }
 
   inline ::std::string genAsm_x86_64() const {
-    ::std::stringstream asmCode;
-
-    asmCode << "section .data" << ENDL;
-
-    asmCode << ENDL;
-    asmCode << "section .rodata" << ENDL;
-
-    asmCode << ENDL;
-    asmCode << "section .bss" << ENDL;
-
-    asmCode << ENDL;
-    asmCode << "section .text" << ENDL;
-    asmCode << INDENT << "global _start" << ENDL;
-    // asmCode << INDENT << "global print:function" << ENDL; // when generating stdlibc
-
-    asmCode << ENDL;
-    asmCode << "_start:" << ENDL;
-    asmCode << INDENT << "call main" << ENDL;
-    asmCode << INDENT
-            << "mov rax, 60                  ; Syscall number for exit (60)"
-            << ENDL;
-    asmCode << INDENT
-            << "mov rdi, rbx                 ; Exit code (0) expects return of "
-               "main to be put in rbx for now"
-            << ENDL;
-    asmCode << INDENT << "syscall                      ; Make the syscall"
-            << ENDL;
-
+    CodeGeneratorEvaluator evaluator;
+   
     for (auto &func : functions) {
-      func.genCode(asmCode);
+      func.generate(evaluator);
```

Comment:
right now the generate function of all nodes seems to be platform dependant, but I suspect your goal is to make multiple CodeGeneratorEvaluator classes for different asm flavours/target platforms, correct? In the opposite case, these functions were intended to be named genAsm_x86_64 like the one we're in right here